src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4918,15 +4916,15 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
   }
 
   QgsFeatureRequest request; // TODO only required attributes
-  if ( values.length() > 2 ) { //filter cached features
-      QString filterString;
-      node = QgsExpressionUtils::getNode( values.at( 2 ), parent );
-      ENSURE_NO_EVAL_ERROR
-      filterString = node->dump();
-      if ( filterString != "NULL" ) {
-          request.setFilterExpression (filterString);
-      }
-  }
+  QString filterString;
+  node = QgsExpressionUtils::getNode( values.at( 2 ), parent );
+  ENSURE_NO_EVAL_ERROR
+  filterString = node->dump();
+  if ( filterString != "NULL" ) request.setFilterExpression (filterString); //filter cached features
+
+  int limit = QgsExpressionUtils::getIntValue( values.at( 3 ), parent ); //in expressions overlay functions throw the exception: Eval Error: Cannot convert '' to int
+  if (limit > 0) request.setLimit (limit);
```

Comment:
I'd remove this check and set the default value for the limit to `-1` (See https://qgis.org/api/classQgsFeatureRequest.html#aa724a450498eeba7a783ead6b62a2e67).